src/test/java/org/reaktivity/nukleus/maven/plugin/internal/generated/FlatFWTest.java

```diff
@@ -133,165 +212,171 @@ public void shouldFailToSetString1BeforeFixed1() throws Exception
-               .set("value1", UTF_8)
-               .build();
-        builder.string1(value)
-               .fixed3(30);
-        value = stringRW.wrap(valueBuffer,  0, valueBuffer.capacity())
-               .set("value2", UTF_8)
-               .build();
-        builder.string2(value)
-               .build();
-        flatRO.wrap(buffer,  0,  100);
-        assertEquals(10, flatRO.fixed1());
-        assertEquals(20, flatRO.fixed2());
-        assertEquals("value1", flatRO.string1().asString());
-        assertEquals(30, flatRO.fixed3());
-        assertEquals("value2", flatRO.string2().asString());
+        return builder.fixed1(10)
+                .string1("value1")
+                .string2("value2")
+                .build()
+                .limit();
     }
 
-    @Test
-    public void shouldSetStringValuesUsingBuffer() throws Exception
+    static int setAllRequiredBufferValues(MutableDirectBuffer buffer, int offset)
     {
-        valueBuffer.putStringWithoutLengthUtf8(0, "value1");
-        valueBuffer.putStringWithoutLengthUtf8(10, "value2");
-        flatRW.wrap(buffer, 0, buffer.capacity())
-            .fixed1(10)
-            .fixed2(20)
-            .string1(valueBuffer, 0, 6)
-            .fixed3(30)
-            .string2(valueBuffer, 10, 6)
-            .build();
-        flatRO.wrap(buffer,  0,  100);
-        assertEquals(10, flatRO.fixed1());
-        assertEquals(20, flatRO.fixed2());
-        assertEquals("value1", flatRO.string1().asString());
-        assertEquals(30, flatRO.fixed3());
-        assertEquals("value2", flatRO.string2().asString());
+        buffer.putLong(offset, 10);
+        buffer.putShort(offset += 8, (short) 222);
+        buffer.putByte(offset += 2, (byte) "value1".length());
+        buffer.putBytes(offset += 1, "value1".getBytes(UTF_8));
+        buffer.putInt(offset+=6, 333);
+        buffer.putByte(offset += 4, (byte) "value2".length());
+        buffer.putBytes(offset += 1, "value2".getBytes(UTF_8));
+
+        return offset + 6;
     }
 
-    @Test
-    public void shouldSetStringValuesToNull() throws Exception
+    static int setAllBufferValues(MutableDirectBuffer buffer, int offset)
     {
-        flatRW.wrap(buffer, 0, buffer.capacity())
-            .fixed1(10)
-            .fixed2(20)
-            .string1((String) null);
+        buffer.putLong(offset, 10);
+        buffer.putShort(offset += 8, (short) 20);
+        buffer.putByte(offset += 2, (byte) "value1".length());
+        buffer.putBytes(offset += 1, "value1".getBytes(UTF_8));
+        buffer.putInt(offset+=6, 30);
+        buffer.putByte(offset += 4, (byte) "value2".length());
+        buffer.putBytes(offset += 1, "value2".getBytes(UTF_8));
+
+        return offset + 6;
     }
 
-    @Test
-    public void shouldSetStringValuesToEmptyString() throws Exception
+    static FlatFW.Builder setAllRequiredBuilderFields(FlatFW.Builder builder, int fieldIndex)
```

Comment:
Rename to setRequiredFields
Parameter name: toFieldIndex
Use if statements like `if (toFieldIndex >= INDEX_STRING1)` to avoid having to duplicate code (and make generation easier and allow passing in Integer.MAX_VALUES)